File: Languages/PHP/Day027.php
<!-- Languages - PHP - Day 27 -->
<!-- Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables. -->

<!-- Create a function that counts how many vowels are in a given string.-->


<?php
// Function to count vowels in a given string using associative array for faster lookup
function count_vowels($input) {
    $vowels = array_flip(str_split('aeiou'));
    $count = 0;
    $input = strtolower($input);

    for ($i = 0; $i < strlen($input); $i++) {
        if (isset($vowels[$input[$i]])) {
            $count++;
        }
    }
    return $count;
}

while (true) {
    echo "Enter a string to count vowels (or type 'exit' to quit): ";
    $input = trim(fgets(STDIN));

    if (strtolower($input) === 'exit') {
        echo "Exiting program. Goodbye!\n";
        break;
    }

    if (empty($input)) {
        echo "Input cannot be empty. Please enter a valid string.\n";
        continue;
    }

    $vowel_count = count_vowels($input);
    echo "Number of vowels in the given string: $vowel_count\n";
}
?>

File: Languages/C/Day027.c
// Languages - C - Day 27
// Foundational Concepts: Basics, syntax, loops, conditionals, functions, and variables.

// Create a function that counts how many vowels are in a given string.


#include <stdio.h>
#include <string.h>
#include <ctype.h>

// Function to count vowels in a given string
int countVowels(const char *input)
{
    int count = 0;
    while (*input)
    {
        char ch = tolower(*input);
        if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u')
        {
            count++;
        }
        input++;
    }
    return count;
}

int main()
{
    char input[100];
    char choice;

    do
    {
        printf("Enter a string to count the number of vowels: ");
        fgets(input, sizeof(input), stdin);

        // Remove newline character
        input[strcspn(input, "\n")] = 0;

        if (strlen(input) == 0)
        {
            printf("Invalid input. Please enter a non-empty string.\n");
            continue;
        }

        int vowelCount = countVowels(input);
        printf("The number of vowels in the given string is: %d\n", vowelCount);

        printf("Would you like to try another string? (y/n): ");
        scanf(" %c", &choice);
        while (getchar() != '\n')
            ; // Clear buffer

    } while (choice == 'y' || choice == 'Y');

    printf("Exiting program. Goodbye!\n");
    return 0;
}
